logic/src/androidMain/kotlin/com/wire/kalium/logic/CoreLogic.kt

```diff
@@ -37,10 +40,25 @@ actual class CoreLogic(
 
             val workScheduler = WorkScheduler(applicationContext, session)
             val syncManager = SyncManagerImpl(workScheduler)
-            AuthenticatedDataSourceSet(networkContainer, proteusClient, workScheduler, syncManager).also {
+            val encryptedSettingsHolder = EncryptedSettingsHolder(applicationContext, "${PREFERENCE_FILE_PREFIX}-${session.userId}")
+            val userPreferencesSettings = KaliumPreferencesSettings(encryptedSettingsHolder.encryptedSettings)
+            val database = Database(applicationContext, "main.db", userPreferencesSettings)
```

Comment:
One more improvement, we could use lazy property to initialise the db
`  val database: Database by lazy { Database(applicationContext, "main.db", userPreferencesSettings) }
`